pydomcfg/tests/bathymetry.py

```diff
@@ -142,34 +142,42 @@ def _calc_rmax(depth):
     Calculate rmax: measure of steepness
 
     This function returns the slope steepness criteria rmax, which is simply
-    (H[0] - H[1]) / (H[0] + H[1])
+    |H[0] - H[1]| / (H[0] + H[1])
 
     Parameters
     ----------
     depth: float
-            Bottom depth (units: m).
+        Bottom depth (units: m).
 
     Returns
     -------
     rmax: float
-            Slope steepness value (units: None)
+        Slope steepness value (units: None)
+
+    Notes
+    -----
+    This function uses a "conservative approach" and rmax is overestimated.
+    rmax at T points is the maximum rmax estimated at any adjacent U/V point.
     """
-    depth = depth.reset_index(list(depth.dims))
 
+    # Loop over x and y
     both_rmax = []
     for dim in depth.dims:
 
-        # (H[0] - H[1]) / (H[0] + H[1])
-        depth_diff = depth.diff(dim)
-        depth_rolling_sum = depth.rolling({dim: 2}).sum().dropna(dim)
-        rmax = depth_diff / depth_rolling_sum
+        # Compute rmax
+        rolled = depth.rolling({dim: 2}).construct("window_dim")
+        diff = rolled.diff("window_dim").squeeze("window_dim")
+        rmax = np.abs(diff) / rolled.sum("window_dim")
 
-        # (R[0] + R[1]) / 2
-        rmax = rmax.rolling({dim: 2}).mean().dropna(dim)
+        # Construct dimension with velocity points adjacent to any T point
+        # We need to shift as we rolled twice and to force boundaries = NaN
+        rmax = rmax.rolling({dim: 2}).construct("vel_points")
+        rmax = rmax.shift({dim: -1})
+        rmax[{dim: [0, -1]}] = None
 
-        # Fill first row and column
-        rmax = rmax.pad({dim: (1, 1)}, constant_values=0)
+        both_rmax.append(rmax)
 
-        both_rmax.append(np.abs(rmax))
+    rmax = xr.concat(both_rmax, "vel_points")
```

Comment:
I think this shows pretty well how `xarray` can be used for this kind of operations.
We created an additional dimension `"vel_points"` with size 4, which is the dimension representing the U/V points adjacent to any T point.